sdk/identity/identity/test/node/deviceCodeCredential.spec.ts

```diff
@@ -242,32 +253,41 @@ describe("DeviceCodeCredential", function() {
     });
   });
 
-  it("cancels polling when abort signal is raised", async function() {
-    const mockHttpClient = new MockAuthHttpClient({
-      authResponse: [
-        { status: 200, parsedBody: deviceCodeResponse },
-        { status: 400, parsedBody: pendingResponse },
-        { status: 400, parsedBody: pendingResponse },
-        { status: 200, parsedBody: { access_token: "token", expires_in: 5 } }
-      ]
+  describe("tests with delays", function() {
+    let delayController: DelayController;
+    before(() => {
+      delayController = createDelayController();
     });
 
-    const credential = new DeviceCodeCredential(
-      "tenant",
-      "client",
-      (details) => assert.equal(details.message, deviceCodeResponse.message),
-      mockHttpClient.identityClientOptions
-    );
-
-    const abortController = new AbortController();
-    const getTokenPromise = credential.getToken("scope", { abortSignal: abortController.signal });
-    await delay(1500); // Long enough for device code request and one polling request
-    abortController.abort();
-
-    const token = await getTokenPromise;
+    it("cancels polling when abort signal is raised", async function() {
+      const mockHttpClient = new MockAuthHttpClient({
+        authResponse: [
+          { status: 200, parsedBody: deviceCodeResponse },
+          { status: 400, parsedBody: pendingResponse },
+          { status: 400, parsedBody: pendingResponse },
+          { status: 200, parsedBody: { access_token: "token", expires_in: 5 } }
+        ]
+      });
+
+      const credential = new DeviceCodeCredential(
+        "tenant",
+        "client",
+        (details) => assert.equal(details.message, deviceCodeResponse.message),
+        mockHttpClient.identityClientOptions
+      );
 
-    assert.strictEqual(token, null);
-    assert.strictEqual(mockHttpClient.requests.length, 2);
+      const abortController = new AbortController();
+      const getTokenPromise = credential.getToken("scope", { abortSignal: abortController.signal });
+      let delay = await delayController.waitForDelay();
```

Comment:
Might be useful to leave a comment here to explain that this code is requesting the promise returned by each `delay()` call in `DeviceCodeCredential` so that the test has control over when they get resolved